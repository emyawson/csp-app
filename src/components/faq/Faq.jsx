import React from 'react';
import { phrase } from '../../utils/phrase';
import './Faq.scss';

const currentEnv = process.env.NODE_ENV;

const Faq = () => (
<div className='faq'>
  <h2 className='faq__headline'>{phrase.searchHeadline}</h2>
    <div data-cytest='faq'>
      <km-search-integration lang='nl-NL'></km-search-integration>
      {currentEnv === 'development' ? <km-faq-accordion lazyLoad="true" listTitle="" widget="m13" faqLang="en"></km-faq-accordion> : <km-faq-accordion lazyLoad="true" listTitle="" widget="m6" faqLang="nl"></km-faq-accordion>}
    </div>
</div>
);

export default Faq;
