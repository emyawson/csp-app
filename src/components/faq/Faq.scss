@import "../../index";

.faq__headline {
    font-size: 18px;
    margin-bottom: 30px;
    margin-top: 43px;
    @include respond(tablet) {
        font-size: 25px;
        margin-bottom: 43px;
        margin-top: 53px;
    }
}

.faq-search {
    display: block;
    width: 100%;
    margin-bottom: 36px;
}