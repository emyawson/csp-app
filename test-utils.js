import React from 'react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import { Router } from 'react-router-dom';
import { connectRouter } from 'connected-react-router';
import { render } from '@testing-library/react';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import { history } from './src/store';
import reducers from './src/store/reducer/index';

const createRootReducer = (reducerHistory) => combineReducers({
  router: connectRouter(reducerHistory),
  ...reducers,
});

// Setup redux persist
const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
};
const persistedReducer = persistReducer(
  persistConfig,
  createRootReducer(history),
);

const renderWithRouter = (ui, options = {}, initialState) => ({
  ...render(
      <Provider store={createStore(persistedReducer, initialState)}>
        <Router history={history}>{ui}</Router>
      </Provider>,
  ),
});

const AllTheProviders = () => {};
const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, renderWithRouter };
