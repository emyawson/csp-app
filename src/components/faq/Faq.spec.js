import React from 'react';
// import { render, screen } from "@testing-library/react";
import Faq from './Faq.jsx';
// import { renderWithRouter } from "../test-utils";
import { renderWithRouter } from '../../../test-utils';

const defaultState = {};

const renderComponent = (state = {}) => {
  const combinedState = {
    ...defaultState,
    ...state,
  };
  return renderWithRouter(<Faq />, combinedState);
};

test('renders <Faq/>', () => {
  const { container } = renderComponent();
  expect(container).toMatchSnapshot();

  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});
