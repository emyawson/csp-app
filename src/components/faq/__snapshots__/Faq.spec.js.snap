// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Faq/> 1`] = `
<div>
  <div
    class="faq"
  >
    <h2
      class="faq__headline"
    >
      Zoeken naar informatie
    </h2>
    <div
      data-cytest="faq"
    >
      <km-search-integration
        lang="nl-NL"
      />
      <km-faq-accordion
        faqlang="nl"
        lazyload="true"
        listtitle=""
        widget="m6"
      />
    </div>
  </div>
</div>
`;
